src/features/vault.ts

```diff
@@ -0,0 +1,247 @@
+import * as vscode from 'vscode';
+import * as utilAnsibleCfg from './utils/ansibleCfg';
+import * as tmp from 'tmp';
+import * as cp from 'child_process';
+import * as fs from 'fs';
+
+async function askForVaultId(ansibleCfg: string) {
+  let vaultId = 'default';
+  let identityList: string[] | undefined;
+  if (ansibleCfg === 'ANSIBLE_VAULT_IDENTITY_LIST') {
+    const envVar: string | undefined = process.env.ANSIBLE_VAULT_IDENTITY_LIST;
+    if (envVar !== undefined) {
+      identityList = envVar
+        .split(',')
+        .map((id: string) => id.split('@', 2)[0].trim());
+    } else {
+      identityList = [];
+    }
+  } else {
+    const cfg: utilAnsibleCfg.AnsibleVaultConfig | undefined =
+      utilAnsibleCfg.getValueByCfg(ansibleCfg);
+    if (cfg !== undefined) {
+      identityList = cfg.defaults.vault_identity_list
+        .split(',')
+        .map((id: string) => id.split('@', 2)[0].trim());
```

Comment:
I noticed that this transformation is copy-pasted in two places, so it's better to just apply it after the whole if-else block.